pkg/skaffold/sync/sync.go

```diff
@@ -140,35 +140,32 @@ func Perform(ctx context.Context, image string, files map[string]string, cmdFn f
 		return errors.Wrap(err, "getting k8s client")
 	}
 
+	numSynced := 0
 	for _, ns := range namespaces {
 		pods, err := client.CoreV1().Pods(ns).List(meta_v1.ListOptions{})
 		if err != nil {
 			return errors.Wrap(err, "getting pods for namespace "+ns)
 		}
 
-		synced := map[string]bool{}
-
 		for _, p := range pods.Items {
 			for _, c := range p.Spec.Containers {
 				if c.Image != image {
 					continue
 				}
 
-				for src, dst := range files {
-					cmds := cmdFn(ctx, p, c, src, dst)
-					for _, cmd := range cmds {
-						if err := util.RunCmd(cmd); err != nil {
-							return err
-						}
+				cmds := cmdFn(ctx, p, c, files)
+				for _, cmd := range cmds {
+					if err := util.RunCmd(cmd); err != nil {
+						return err
 					}
-					synced[src] = true
+					numSynced++
 				}
 			}
 		}
+	}
 
-		if len(synced) != len(files) {
-			return errors.New("couldn't sync all the files in " + ns)
-		}
+	if numSynced == 0 {
```

Comment:
Not sure if this check makes sense anymore?

Reply:
I put this in to pass the "no copy" test at `sync_test.go`:410.  If you want, both this and the test can be removed.